### PR TITLE
raw_spacetime_lib refactor + related fixes

### PR DESCRIPTION
**Pull request by nojb**
I am trying to get Spacetime to work on Windows. As part of this, I did some refactorings of `raw_spacetime_lib` that have independent interest, which I am submitting in this PR. Along the way, it seems I found a bug in the handling of the deprecated `"noalloc"` attribute.

- Move `spacetime_offline.c` (which is used to read the Spacetime profiles) out of the runtime system and into the library `raw_spacetime_lib`.
- Use standard `otherlibs/Makefile` to simplify `otherlibs/raw_spacetime_lib/Makefile`
- Allow the primitives in `spacetime_offline.c` to be used in bytecode as well.  Previously in bytecode these primitives were short-circuited to fail, but I don't see any reason to do so and this allows to use `raw_spacetime_lib` from e.g. the toplevel.
- Replace `"noalloc"` by `[@@noalloc]` in `raw_spacetime_lib`

This last change revealed that the deprecated attributes `"noalloc"`, `"float"` were being ignored when appearing *after* a pair of names `"bytecode primitive" "native-code primitive"`: see the commit "Fix deprecated noalloc handling".

I experimented a little bit with `raw_spacetime_lib` from the toplevel and it seems to work, but would appreciate a confirmation that these primitives are safe for bytecode from someone more knowledgeable about Spacetime internals.

@mshinwell @lpw25

### CODE REVIEW

File: otherlibs/raw_spacetime_lib/raw_spacetime_lib.ml [hunk 1]
```diff
@@ -496,8 +472,7 @@ module Trace = struct
 
     (* CR-soon lwhite: These functions should work in bytecode *)
```
Comment (mshinwell) [inline otherlibs/raw_spacetime_lib/raw_spacetime_lib.ml]: Please delete this `CR-soon` line.

File: typing/primitive.ml [hunk 2]
```diff
@@ -85,7 +85,9 @@ let parse_declaration valdecl ~native_repr_args ~native_repr_res =
   let arity = List.length native_repr_args in
   let name, native_name, old_style_noalloc, old_style_float =
     match valdecl.pval_prim with
+    | name :: name2 :: "noalloc" :: "float" :: _
```
Comment (mshinwell) [inline typing/primitive.ml]: What are these changes needed for?  I think these should go in a separate GPR.

File: otherlibs/raw_spacetime_lib/raw_spacetime_lib.ml [hunk 3]
```diff
@@ -601,7 +576,7 @@ module Heap_snapshot = struct
       call_counts : bool;
     }
 
-    let pathname_suffix_trace = "trace"
+    let _pathname_suffix_trace = "trace"
```
Comment (mshinwell) [inline otherlibs/raw_spacetime_lib/raw_spacetime_lib.ml]: Here and elsewhere, let's delete the unused values / functions.

**Review by mshinwell — APPROVED** [review]
OK after minor changes addressed.

File: typing/primitive.ml (hunk 2 above)
Comment (nojb) [reply typing/primitive.ml]: This fixes a bug in the handling of deprecated "noalloc" and "float" attributes (which is triggered by `raw_spacetime_lib.ml` as it exists on `trunk`), but indeed they are not needed in this PR; I will put it in a separate one.

**Comment by nojb** [comment]
Thanks for the review @mshinwell! Amended as suggested.

**Comment by mshinwell** [comment]
OK, I will merge.  (By the way, I'm not sure if you rewrote existing changesets or something, but I wasn't able to see just the changes since my review.  You might want to check your git workflow.)

**Comment by nojb** [comment]
I rebased on `trunk`, sorry about that!
